www/front_src/src/Resources/api/searchObjects.ts

```diff
@@ -0,0 +1,57 @@
+type SearchableField =
+  | 'host.name'
+  | 'host.alias'
+  | 'host.address'
+  | 'service.description';
+
+const searchOptions: Array<SearchableField> = [
+  'host.name',
+  'host.alias',
+  'host.address',
+  'service.description',
+];
+
+interface SearchObject {
+  field: SearchableField;
+  value: string;
+}
+
+interface SearchParam {
+  $or: Array<{ [field in SearchableField]?: { $lk: string } }>;
+}
+
+const getFoundSearchObjects = (searchValue: string): Array<SearchObject> => {
+  const searchOptionMatches = searchOptions.map((searchOption) => {
+    const pattern = `${searchOption.replace('.', '\\.')}:([^\\s]+)`;
```

Comment:
there will be search query like host.name:host_\d+
Then, need to check that this line does not break regular expression

Reply:
`[^\\s]+` indicates any non-whitespace character, so we're basically sending the search value as-is. This should still work with regex-like query